apis/mattermost/v1beta1/mattermost_types.go

```diff
@@ -204,6 +208,16 @@ type Probes struct {
 	ReadinessProbe v1.Probe `json:"readinessProbe,omitempty"`
 }
 
+// PodTemplate defines configuration for the template for Mattermost pods.
+type PodTemplate struct {
+	// Defines the security context for the Mattermost app server pods.
+	// +optional
+	SecurityContext v1.PodSecurityContext `json:"securityContext,omitempty"`
+	// Defines the security context for the Mattermost app server container.
+	// +optional
+	ContainerSecurityContext v1.SecurityContext `json:"containerSecurityContext,omitempty"`
```

Comment:
I think we should use pointers for those to clearly indicate the absence, otherwise this will be automatically initialized to an empty struct, bringing all the zero values along, which in turn might result in some unintended behaviour.

Reply:
Updated, let me know if there's anything else (or if I changed things incorrectly 😁 ).